lib/ts/recipe/dashboard/api/getTenantLoginMethodsInfo.ts

```diff
@@ -0,0 +1,141 @@
+    };
+    thirdParty: {
+        enabled: boolean;
+    };
+};
+
+export type Response = {
+    status: "OK";
+    tenants: TenantLoginMethodType[];
+};
+
+export default async function getTenantLoginMethodsInfo(
+    _: APIInterface,
+    __: string,
+    ___: APIOptions,
+    userContext: any
+): Promise<Response> {
+    const tenantsRes = await Multitenancy.listAllTenants(userContext);
+    const finalTenants: TenantLoginMethodType[] = [];
+
+    if (tenantsRes.status !== "OK") {
+        return tenantsRes;
+    }
+
+    for (let i = 0; i < tenantsRes.tenants.length; i++) {
+        const currentTenant = tenantsRes.tenants[i];
+
+        const normalisedTenantLoginMethodsInfo = normaliseTenantLoginMethodsWithInitConfig({
+            tenantId: currentTenant.tenantId,
+            emailPassword: currentTenant.emailPassword,
+            passwordless: currentTenant.passwordless,
+            thirdParty: currentTenant.thirdParty,
+        });
+
+        finalTenants.push(normalisedTenantLoginMethodsInfo);
+    }
+
+    return {
+        status: "OK",
+        tenants: finalTenants,
+    };
+}
+
+function normaliseTenantLoginMethodsWithInitConfig(
+    tenantDetailsFromCore: TenantLoginMethodType
+): TenantLoginMethodType {
+    const normalisedTenantLoginMethodsInfo: TenantLoginMethodType = {
+        tenantId: tenantDetailsFromCore.tenantId,
+        emailPassword: {
+            enabled: false,
+        },
+        thirdPartyEmailPasssword: {
+            enabled: false,
+        },
+        passwordless: {
+            enabled: false,
+        },
+        thirdPartyPasswordless: {
+            enabled: false,
+        },
+        thirdParty: {
+            enabled: false,
+        },
+    };
+
+    if (tenantDetailsFromCore.passwordless.enabled === true) {
+        try {
+            const thirdpartyPasswordlessRecipe = ThirdPartyPasswordlessRecipe.getInstanceOrThrowError();
+            normalisedTenantLoginMethodsInfo.thirdPartyPasswordless!.enabled = true;
+            normalisedTenantLoginMethodsInfo.thirdPartyPasswordless!.contactMethod =
+                thirdpartyPasswordlessRecipe.config.contactMethod;
+        } catch (_) {
+            try {
+                const passwordlessRecipe = PasswordlessRecipe.getInstanceOrThrowError();
+                normalisedTenantLoginMethodsInfo.passwordless.enabled = true;
+                normalisedTenantLoginMethodsInfo.passwordless.contactMethod = passwordlessRecipe.config.contactMethod;
+            } catch (_) {}
+        }
+    }
+
+    if (tenantDetailsFromCore.emailPassword.enabled === true) {
+        try {
+            ThirdPartyEmailPasswordRecipe.getInstanceOrThrowError();
+            normalisedTenantLoginMethodsInfo.thirdPartyEmailPasssword!.enabled = true;
+        } catch (_) {
+            try {
+                EmailPasswordRecipe.getInstanceOrThrowError();
+                normalisedTenantLoginMethodsInfo.emailPassword.enabled = true;
+            } catch (_) {}
```

Comment:
this should go in its own block... users can theoritically have both EmailPassword and ThirdPartyEmailPasswordRecipe enabled!

Reply:
done